tests/test_peer_heap.py

```diff
@@ -107,6 +107,57 @@ def test_update(peer_heap, peers):
     assert peer_heap.peek_peer() == p
 
 
+def test_remove(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = len(peers)
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        peer_heap.remove_peer(p)
+        verify(peer_heap, 0)
+
+
+def test_remove_duplicate(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = random.randint(0, len(peers))
+
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        peer_heap.remove_peer(p)
+        peer_heap.remove_peer(p)
+        verify(peer_heap, 0)
+
+
+def test_remove_from_empty_heap():
+    heap = PeerHeap()
+    heap.remove_peer(mock_peer())
+
+
+def test_remove_none_exist(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    peer_heap.remove_peer(peers[0])
+
+
+def test_remove_mismatch(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    # create a fake peer with duplicated index.
+    fake_peer = mock_peer()
+    fake_peer.index = 1
+    with pytest.raises(AssertionError):
```

Comment:
this is not valid case. same score doesn't mean same index. Remove peer is based on the index.
That's why I write it by changing the index.